src/object/object_primitive.c

```diff
@@ -16374,3 +16377,224 @@ mr_get_compression_length (const char *string, int charlen)
+
+  /* Alloc memory for the compressed string */
+  /* Worst case LZO compression size from their FAQ */
+  compressed_string = malloc (str_length + (str_length / 16) + 64 + 3);
+  if (compressed_string == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY,
+	      1, (size_t) (str_length + (str_length / 16) + 64 + 3));
+      goto cleanup;
+    }
+
+  /* Compress the string */
+  rc = lzo1x_1_compress ((lzo_bytep) string, (lzo_uint) str_length, (lzo_bytep) compressed_string, &compression_length,
+			 wrkmem);
+  if (rc != LZO_E_OK)
+    {
+      /* We should not be having any kind of errors here. Because if this compression fails, there is not warranty
+       * that the compression from putting data into buffer will fail as well. This needs to be checked but for now
+       * we leave it as it is.
+       */
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_LZO_COMPRESS_FAIL, 4, FILEIO_ZIP_LZO1X_METHOD,
+	      fileio_get_zip_method_string (FILEIO_ZIP_LZO1X_METHOD), FILEIO_ZIP_LZO1X_DEFAULT_LEVEL,
+	      fileio_get_zip_level_string (FILEIO_ZIP_LZO1X_DEFAULT_LEVEL));
+      goto cleanup;
+    }
+
+  if (compression_length < (lzo_uint) (str_length - 8))
+    {
+      /* Compression successful */
+      length = (int) compression_length;
+      compressed = true;
+      str = compressed_string;
+    }
+  else
+    {
+      /* Compression failed */
+      length = str_length;
+      compression_length = 0;
+      str = string;
+    }
+
+  /* 
+   * Step 2 : Compute the disk size of the dbvalue.
+   * We are sure that the initial string length is greater than 255, which means that the new encoding applies.
+   */
+
+  switch (DB_VALUE_DOMAIN_TYPE (value))
+    {
+    case DB_TYPE_VARCHAR:
+    case DB_TYPE_VARNCHAR:
+      *val_size = or_packed_varchar_length (length + PRIM_TEMPORARY_DISK_SIZE) - PRIM_TEMPORARY_DISK_SIZE;
+      break;
+
+    default:
+      /* It should not happen */
+      assert (false);
+      rc = ER_FAILED;
+      goto cleanup;
+    }
+
+  /* Step 3 : Initialize the buffer */
+  OR_BUF_INIT (*buf, val_p, *val_size);
+
+  /* Step 4 : Insert the disk representation of the dbvalue in the buffer */
+
+  switch (DB_VALUE_DOMAIN_TYPE (value))
+    {
+    case DB_TYPE_VARNCHAR:
+    case DB_TYPE_STRING:
+      rc = mr_write_compressed_string_to_buffer (buf, str, (int) compression_length, str_length, align);
+      break;
+
+    default:
+      /* It should not happen. */
+      assert (false);
+      goto cleanup;
+    }
+
+cleanup:
+  buf->error_abort = error_abort;
+
+  if (compressed_string != NULL)
+    {
+      free_and_init (compressed_string);
+    }
+
+  if (wrkmem != NULL)
+    {
+      free_and_init (wrkmem);
+    }
+
+  if (rc == ER_TF_BUFFER_OVERFLOW)
+    {
+      return or_overflow (buf);
+    }
+
+  return rc;
+}
+
+/* mr_write_compressed_string_to_buffer()	  : Similar function to the previous implementation of 
+ *						    or_put_varchar_internal.
```

Comment:
In function 'or_put_varchar_internal', compressable is set to true when charlen > 0xff; it should be when charlen >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION (Now, it has the same value, but the code is not safe if we change it).